evaluate/evaluate_test.go

```diff
@@ -318,19 +318,16 @@ func TestEvaluate(t *testing.T) {
 		d = bytes.ReplaceAll(d, []byte("plain"), []byte("next"))
 		require.NoError(t, os.WriteFile(repositoryNextConfigPath, d, 0))
 
-		generateTestsForFilePlainSuccess := func(args mock.Arguments) {
-			require.NoError(t, os.WriteFile(filepath.Join(args.String(2), "plain_test.go"), []byte("package plain\nimport \"testing\"\nfunc TestFunction(t *testing.T){}"), 0600))
-		}
 		generateTestsForFilePlainSuccessMetrics := metrics.Assessments{
 			metrics.AssessmentKeyProcessingTime: 1,
 		}
 		generateTestsForFilePlainError := errors.New("generateTestsForFile error")
 
 		generateSuccess := func(mockedModel *modeltesting.MockModel) {
-			mockedModel.On("GenerateTestsForFile", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(generateTestsForFilePlainSuccessMetrics, nil).Run(generateTestsForFilePlainSuccess).Once()
+			mockedModel.RegisterGenerateSuccess(t, "plain_test.go", "package plain\nimport \"testing\"\nfunc TestFunction(t *testing.T){}", generateTestsForFilePlainSuccessMetrics).Once()
```

Comment:
The plain test file name and content are always the same. Can we extract those into a struct inside this test file?

```go
testFile struct {
  Path string
  Content string
}

testFiles := map[string]testFile
```

`RegisterGenerateSuccess(t, testFiles["plain"].Path, testFiles["plain"].Content, `

I guess it will otherwise be copied more and more as soon as we are adding additional tests for additional functionality.